simphony_mayavi/add_dataset.py

```diff
@@ -0,0 +1,65 @@
+from mayavi import mlab
+
+from simphony.cuds.abc_mesh import ABCMesh
+from simphony.cuds.abc_particles import ABCParticles
+from simphony.cuds.abc_lattice import ABCLattice
+
+from simphony_mayavi.sources.api import CUDSSource
+
+
+def add_dataset(cuds, point_scalars=None, point_vectors=None,
+                cell_scalars=None, cell_vectors=None):
+    ''' Add CUDS dataset to the current scene.
+    If no scene is defined, a new scene is created.
+
+    Parameters
+    ----------
+    cuds : ABCMesh, ABCParticles, ABCLattice
+
+    point_scalars : str, optional
+        CUBA name of the data to be selected as point scalars.
+        Default is the first available point scalars.
+
+    point_vectors : str, optional
+        CUBA name of the data to be selected as point vectors.
+        Default is the first available point vectors.
+
+    cell_scalars : str, optional
+        CUBA name of the data to be selected as cell scalars.
+        Default is the first available cell scalars.
+
+    cell_vectors : str, optional
+        CUBA name of the data to be selected as cell vectors.
+        Default is the first available cell vectors.
+
+    Returns
+    -------
+    source : CUDSSource
+
+    Notes
+    -----
+    To turn off visualisation of a point/cell data, assign
+    [point/cell]_[scalars/vectors] to an empty string
+    '''
+    if isinstance(cuds, (ABCMesh, ABCParticles, ABCLattice)):
```

Comment:
I see this pattern quite often, and I was wondering if it would make sense to have a CUDSSource.from_cuds() classmethod. The only potential problem is that it might introduce some circular dependencies. Or maybe move the check in the CUDSSource constructor?